go/teams/teams.go

```diff
@@ -693,7 +693,33 @@ func (t *Team) sigTeamItem(ctx context.Context, section SCTeamSection, linkType
 	return sigMultiItem, nil
 }
 
-func (t *Team) recipientBoxes(ctx context.Context, memSet *memberSet) (*PerTeamSharedSecretBoxes, *SCPerTeamKey, error) {
+func (t *Team) recipientBoxes(ctx context.Context, memSet *memberSet) (*PerTeamSharedSecretBoxes, map[keybase1.TeamID]*PerTeamSharedSecretBoxes, *SCPerTeamKey, error) {
+
+	// get device key
+	deviceEncryptionKey, err := t.G().ActiveDevice.EncryptionKey()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// First create all the subteam per-team-key boxes for new implicit admins.
+	// We'll return these whether or not we're doing a rotation below.
+	// TODO: Should we no-op this if the admins+owners aren't actually new?
```

Comment:
That is, a new admin might already have been keyed for a subteam if they're already a member/otherwise-implicit-admin of it. Is it an error if we post key boxes for them again? If so we'll need to client to check whether they're an admin of any intermediate subteam, and probably we'll want to add a couple test cases for that.